In the Base Rate Neglect notebook (see: https://github.com/DaisyWelham/Statistics-and-Data-Biases/blob/main/Base%20Rate%20Neglect.ipynb ) we saw that if we have 2,500 blobs taking a test with 95% true positive rate, 5% false negative rate, 98% true negative rate, and 2% false negative rate, a single positive test gives us an expected 4.76% chance of actually having the disease (we called it "jeebies"). Suppose we want to do better than this, how can we do it?

The way to update beliefs with new evidence is called Bayes' Theorem. We find the probability of having the disease (X) given the evidence (E) using:

P(X|E) = P(E|X) * P(X) / P(E)

Where P(X) is our "prior", our current best guess of whether we have the disease regardless of evidence, P(E|X) is the probability we would test positive given we have the disease, and P(E) is our total probability of testing positive regardless of if we have the disease or not. 

Let's implement a Bayes Theorem calculator and show that it agrees with our predictions from the Base Rate Neglect notebook!

## Bayes Theorem

In [1]:
def bayesTheorem(probEgivenX, probX, probE):
    return(probEgivenX * probX / probE)

probX = 0.001 #Base rate
probEgivenX = 0.95 #True positive rate

'''
probE comes from the probability that we are positive and test positive, 
plus the probabiltiy that we are not positive but test positive anyway
'''

probE = (0.95 * 0.001)  + (0.999 * 0.02)

print(f"Probability we have jeebies given one positive test according to Bayes' Theorem: {100 * bayesTheorem(probEgivenX, probX, probE)}%" )

Probability we have jeebies given one positive test according to Bayes' Theorem: 4.538939321548017%


Now suppose we do another test, with the same probabilities, but independently of our original test. We test positive again. We can do Bayes' theorem again, with our new prior P(X). What is the probability we have jeebies now?

In [2]:
probX = 0.04538939321548017
#probEgivenX doesn't change
probE = (0.95 * probX)  + (0.999 * 0.02)
print(f"Probability we have jeebies given two positive tests according to Bayes' Theorem: {100 * bayesTheorem(probEgivenX, probX, probE)}%" )

Probability we have jeebies given two positive tests according to Bayes' Theorem: 68.33593628258868%


68.3% is much more confidence! But we want to be really sure, so let's do a third test.

In [3]:
probX = 0.6833593628258868
#probEgivenX doesn't change
probE = (0.95 * probX)  + (0.999 * 0.02)
print(f"Probability we have jeebies given three positive tests according to Bayes' Theorem: {100 * bayesTheorem(probEgivenX, probX, probE)}%" )

Probability we have jeebies given three positive tests according to Bayes' Theorem: 97.01421785827868%


Okay, it really does seem like we have jeebies! Time to stay home and learn more about statistics! :)